Userland/Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -584,6 +586,23 @@ void TextEditorWidget::update_title()
 
 void TextEditorWidget::open_file(const String& path)
 {
+    if (Core::File::exists(path)) {
+        auto guessed_mime = Core::guess_mime_type_based_on_filename(path);
+        if (!guessed_mime.has_value() || guessed_mime->type() != Core::MimeType::Type::text) {
+            StringBuilder error_builder;
+            if (!guessed_mime.has_value())
+                error_builder.append("This file does not have a MIME type, and so it may not be a text file. ");
+            else
+                error_builder.appendff("This file has the MIME type of '{}', and so it may not be a text file. ", guessed_mime->as_string());
+            error_builder.append("Are you sure you want to open this?");
+
+            if (GUI::MessageBox::show(window(), error_builder.to_string(), "Not a text file", GUI::MessageBox::Type::Warning,
+                    GUI::MessageBox::InputType::YesNo)
+                == GUI::Dialog::ExecResult::ExecNo)
+                return;
+        }
```

Comment:
This feels like we're being unnecessarily obnoxious to the user: they've asked us to open a file, and instead of doing that, we first get in their way and say "uh-huh, well are you suuure you want to open this file though?" even though opening a file a non-destructive operation.

Can we find a less obstructive approach? Perhaps a dismissable banner above the text editor widget with some text e.g "NOTE: This file has non-textual MIME type abc/def"

Reply:
Yeah, I was worried it was a bit annoying, and you're right -- it's non-destructive so who cares if we crash!

I like the banner idea, unfortunately I'm not sure if I know how to implement that very well... I'll take a look notifications for some :sparkles:inspiration:sparkles:.